WebAPI/Controllers/CategoriesController.cs

```diff
@@ -22,4 +22,25 @@ public async Task<ActionResult<List<CategoryDetailDto>>> GetAll()
 
         return Ok(result);
     }
+
+    [HttpGet]
+    public async Task<ActionResult<string>> GetImage(Guid id)
+    {
+        var result = await _categoryService.GetByIdAsync(id);
+
+        return Ok(result.ImagePath);
+    }
+
+    [HttpPost]
+    public async Task<ActionResult<Task>> UpdateImage(CategoryDetailDto updatedCategory)
```

Comment:
1. No need to return task result for response, you can just leave `Task<ActionResult>`
2. For updating entities you should use **only** `PUT` or `PATCH` HTTP-methods since this is RESTful API, please, take a look at doc I shared earlier what does HTTP idempotency means in REST.

Same case for `ProductsController`

Reply:
1. But what should I return then? It returns a task

`Cannot implicitly convert type 'System.Threading.Tasks.Task' to 'Microsoft.AspNetCore.Mvc.ActionResult'`

Reply:
Usually, if you update entity, you should return the updated entity as well, so, it would be `Task< ActionResult< CategoryDetailDto >>`.